exercises/practice/simple-cipher/tests/Tests.elm

```diff
@@ -0,0 +1,140 @@
+module Tests exposing (tests)
+
+import Expect
+import Fuzz
+import SimpleCipher
+import Test exposing (Test, describe, fuzz, fuzz2, skip, test)
+
+
+tests : Test
+tests =
+    describe "SimpleCipher"
+        [ describe "Substitution cipher"
+            [ -- skip <|
+              test "Can encode" <|
+                \() ->
+                    SimpleCipher.encode "abcdefghij" "aaaaaaaaaa"
+                        |> Expect.equal "abcdefghij"
+            , skip <|
+                test "Can decode" <|
+                    \() ->
+                        SimpleCipher.decode "abcdefghij" "abcdefghij"
+                            |> Expect.equal "aaaaaaaaaa"
+            , skip <|
+                fuzz2
+                    (Fuzz.intRange (Char.toCode 'a') (Char.toCode 'z')
+                        |> Fuzz.map Char.fromCode
+                        |> Fuzz.list
+                        |> Fuzz.map String.fromList
+                    )
+                    (Fuzz.intRange (Char.toCode 'a') (Char.toCode 'z')
+                        |> Fuzz.map Char.fromCode
+                        |> Fuzz.listOfLengthBetween 1 100
+                        |> Fuzz.map String.fromList
+                    )
+                    "encoding then decoding gives the original message"
+                <|
+                    \message key ->
+                        message
+                            |> SimpleCipher.encode key
+                            |> SimpleCipher.decode key
+                            |> Expect.equal message
```

Comment:
@ceddlyburge this one is a bit verbose because I had to generate string fuzzers with only `a` to `z`, but it's the perfect use of a fuzzer. The error message looks like
```
↓ SimpleCipher
↓ Substitution cipher
✗ encoding then decoding gives the original message

Given ("a","b")

    "c"
    ╷
    │ Expect.equal
    ╵
    "a"
```

Reply:
I agree, this looks great. Maybe create a new fuzzer that generates the key? It's used in enough places to justify it I think.